### PR TITLE
Time improvements

### PR DESCRIPTION
**Pull request by Matt3o12**
tl;dr:

 - all rss/atom parses did not set DateValid correctly at all. This has now been fixed (commit 1)
 - improved `parseTime` to better handle timezones (I hate them too... -- commit 2)
 - More tests!

commit bbde65d3b9d3e67c0dba33b68dca8096aa0bff2d
Author: Matteo Kloiber <info@matt3o12.de>
Date:   Tue May 8 14:57:09 2018 +0200

    parseTime: text timezones are now parsed correctly
    
    If a format contains the timezone as text,
    such as 'Jan 2, 2006 15:04 PM MST',
    golang does not automacitally load the timezone
    offset. This has to be done manually through
    LoadTimezone. The variable
    `TimeLayoutsLoadLocation` contains all parse
    formats with such a timezone. parseTime now
    loads the timezone (if possible) and returns
    the time in the correct timezone.
    
    Tests have also been added to test more time
    formats. TimeLayouts have also been added
    for date that contain both the texual timezone
    and offset (such as Mon Jan _2 15:04:05 MST
    -0200). In this case, parseTime always uses
    the offset, even if it conflicts with the
    known offset because it can be ambiguous:
    https://news.ycombinator.com/item?id=10199812

commit 7389b38786afa0d7e9e18df919117be88cefa614
Author: Matteo Kloiber <info@matt3o12.de>
Date:   Tue May 8 13:36:56 2018 +0200

    DateValid is now set correctly
    
    Due to a bug, DateValid was only set for the
    protocol specific item (rss_1_0, etc), not
    the returned item. This has now been fixed.
    The unit tests have been extended to ensure
    that for at least one item for each protocol.

### CODE REVIEW

**Comment by SlyMarbo** [comment]
This is great stuff; thanks very much for your work.
